examples/frame_prediction/js/entities/entities.js

```diff
@@ -0,0 +1,76 @@
+/************************************************************************************/
+/*                                                                                  */
+/*     Base entity that predicts its position based on the last update              */
+/*                                                                                  */
+/************************************************************************************/
+game.FramePredictionEntity = me.Entity.extend({
+   
+    init: function(x, y, settings) {
+        this._super(me.Entity, "init", [x, y , settings]);
+    },
+    
+    draw: function(renderer) {
+        // Predict the next position to draw at based on velocity
+        var updateInterval = (me.timer.getTime() - me.timer.lastUpdate) / ( (1000/me.sys.updatesPerSecond) );
```

Comment:
For each draw call, the Entity is starting out in the state of time of when it was last updated.

So for the prediction demo, we have 4 draws for every 1 update. Each time you draw, the entity's stored information (x, y, rotation, etc) is still stuck in time 0-3 frames ago. So you need to know the timestamp from when this information was last updated to make a prediction, not the timestamp from the last frame.